alibi_detect/utils/schemas.py

```diff
@@ -1,158 +1,450 @@
-# type: ignore[assignment]
-# Above is needed to avoid mypy throwing up errors due to pydantic model fields being redefined in inherited
-# pydantic models. Could be avoided if we allow-redefinition, or else avoid redefinition (would increase duplication).
+"""
+Pydantic models used by :func:`~alibi_detect.utils.validate.validate_config` to validate configuration dictionaries.
+The `resolved` kwarg of :func:`~alibi_detect.utils.validate.validate_config` determines whether the *unresolved* or
+*resolved* pydantic models are used:
 
-# TODO - conditional checks depending on backend
+- The *unresolved* models expect any artefacts specified within it to not yet have been resolved.
+  The artefacts are still string references to local filepaths or registries (e.g. `x_ref = 'x_ref.npy'`).
+
+- The *resolved* models expect all artefacts to be have been resolved into runtime objects. For example, `x_ref`
+  should have been resolved into an `np.ndarray`.
+
+.. note::
+    For detector pydantic models, the fields match the corresponding detector's args/kwargs. Refer to the
+    detector's api docs for a full description of each arg/kwarg.
+"""
+
+# TODO - conditional checks depending on backend etc
 # TODO - consider validating output of get_config calls
 import numpy as np
 from pydantic import BaseModel
-from typing import Optional, Union, Dict, List, Callable, Any
-from alibi_detect.utils._types import Literal
+from typing import Optional, Union, Dict, List, Callable, Any, Type
+from alibi_detect.utils._types import Literal, NDArray
 from alibi_detect.version import __version__, __config_spec__
 from alibi_detect.models.tensorflow import TransformerEmbedding
-from transformers import PreTrainedTokenizerBase
 from alibi_detect.utils.frameworks import has_tensorflow
-
-# SupportedModels_types is a tuple of possible models (conditional on installed deps). This is used in isinstance() etc.
-SupportedModels = []
+from alibi_detect.cd.tensorflow import UAE as UAE_tf, HiddenOutput as HiddenOutput_tf
+from transformers import PreTrainedTokenizerBase
+SupportedModels_li = []
```

Comment:
`li` suffix isn't very suggestive of anything... :)

Reply:
It was short for `list` but I guess that isn't very clear!